ngraph_creator/operations/src/Add.cpp

```diff
@@ -5,19 +5,46 @@ namespace hardware {
 namespace neuralnetworks {
 namespace nnhal {
 
-Add::Add(int operationIndex) : OperationsBase(operationIndex) {}
+Add::Add(int operationIndex) : OperationsBase(operationIndex) {
+    mDefaultOutputIndex = sModelInfo->getOperationOutput(mNnapiOperationIndex, 0);
+}
+
+bool Add::validate() {
+    // check output type
+    if (!checkOutputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32)) {
+        return false;
+    }
 
-bool Add::validate() { return true; }
+    // Check all input types
+    for (int i = 0; i <= 1; i++) {
+        if (!checkInputOperandType(i, (int32_t)OperandType::TENSOR_FLOAT32)) return false;
+    }
+
+    if (!checkInputOperandType(2, (int32_t)OperandType::INT32)) {
+        return false;
+    }
+
+    return true;
+}
 
 // TODO: Implement APIs createNode & createNodeForPlugin. These are just dummy placeholders.
```

Comment:
Please remove this comment, as you have implemented the Operation APIs now.

Reply:
Also, change the commit for "Add" to "Add operation implementation", as the code was just dummy code earlier.